### ISSUE TITLE
"LinqException: Object initializer expected for insert statement" when inserting into SQLite 

### ISSUE THREAD

**Issue by alimbada** [linq2db/linq2db.LINQPad#119]
```C#
var mailData = new Mail
{
	// Set some properties
};

Mail.Insert(() => mailData);
```

Throws this exception:
```Text
LinqException: Object initializer expected for insert statement.
   at LinqToDB.Linq.Builder.UpdateBuilder.BuildSetterWithContext(ExpressionBuilder builder, BuildInfo buildInfo, LambdaExpression setter, IBuildContext into, List`1 items, IBuildContext[] sequences)
   at LinqToDB.Linq.Builder.UpdateBuilder.BuildSetter(ExpressionBuilder builder, BuildInfo buildInfo, LambdaExpression setter, IBuildContext into, List`1 items, IBuildContext sequence)
   at LinqToDB.Linq.Builder.InsertBuilder.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.Build[T]()
   at LinqToDB.Linq.Query`1.CreateQuery(ExpressionTreeOptimizationContext optimizationContext, ParametersContext parametersContext, IDataContext dataContext, Expression expr)
   at LinqToDB.Linq.Query`1.GetQuery(IDataContext dataContext, Expression& expr, Boolean& dependsOnParameters)
   at LinqToDB.Linq.ExpressionQuery`1.GetQuery(Expression& expression, Boolean cache, Boolean& dependsOnParameters)
   at LinqToDB.Linq.ExpressionQuery`1.System.Linq.IQueryProvider.Execute[TResult](Expression expression)
   at LinqToDB.LinqExtensions.InsertWithIdentity[T](ITable`1 target, Expression`1 setter)
   at LinqToDB.LinqExtensions.InsertWithInt32Identity[T](ITable`1 target, Expression`1 setter)
   at UserQuery.Main() in C:\Users\Ammaar.Limbada\AppData\Local\Temp\LINQPad5\_otgrdadd\query_rncuvn.cs:line 56
   at LINQPad.ExecutionModel.ClrQueryRunner.Run()
   at LINQPad.ExecutionModel.Server.RunQuery(QueryRunner runner)
   at LINQPad.ExecutionModel.Server.StartQuery(QueryRunner runner)
   at LINQPad.ExecutionModel.Server.<>c__DisplayClass151_0.<ExecuteClrQuery>b__0()
   at LINQPad.ExecutionModel.Server.SingleThreadExecuter.Work()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

**Comment by MaceWindu** [comment]
This is expected error. This API works with expressions:
```cs
Mail.Insert(() => new Mail
{
	// Set some properties
});
```

For object-based APIs, use methods on context:
```cs
this.Insert(mailData);
```

**Comment by alimbada** [comment]
Ah, thanks. I think I tried changing it to:

```C#
Mail.DataContext.Insert(() => mailData);
```

but that didn't work as it doesn't take an `Expression`/`Func`

I had to also add a reference to `Microsoft.Bcl.AsyncInterfaces` as it wouldn't compile when using `DataContext.Insert`. For some reason I couldn't find the assembly manually but letting LinqPad add the reference automatically seemed to work.

Thanks again.
